src/parser/ion_1_0/text/mod.rs

```diff
@@ -0,0 +1,2614 @@
+            .map_err(|_| SymbolError::InvalidSid(s.to_string()))?;
+        match table.borrow().lookup_sid(sid) {
+            Ok(token) => Ok(token),
+            Err(e) => Err(e),
+        }
+    } else {
+        make_symbol(table, text)
+    }
+}
+
+fn make_symbol(table: Table, text: String) -> Result<SymbolToken, SymbolError> {
+    let token = SymbolToken::Known { text };
+    table.borrow_mut().add_symbol(&token);
+    Ok(token)
+}
+
+/// Note: IDENTIFIER_SYMBOL covers all cases provided by TYPE.
+///
+/// symbol
+///     : IDENTIFIER_SYMBOL
+///     // note that this is because we recognize the type names for null
+///     // they are ordinary symbols on their own
+///     | TYPE
+///     | QUOTED_SYMBOL
+///     ;
+fn take_symbol(table: Table) -> impl Fn(&str) -> IonResult<&str, SymbolToken> {
+    move |i: &str| {
+        let (i, result) = alt((
+            map(take_identifier_symbol, |s| {
+                get_or_make_symbol(table.clone(), s.to_string())
+            }),
+            map(take_quoted_symbol, |s| make_symbol(table.clone(), s)),
+        ))(i)?;
+
+        match result {
+            Ok(token) => Ok((i, token)),
+            Err(e) => Err(Err::Failure(IonError::from_symbol_error(i, e))),
+        }
+    }
+}
+
+fn take_sid(i: &str) -> IonResult<&str, &str> {
+    preceded(char('$'), take_while1(is_dec_digit))(i)
+}
+
+fn eat_ws(i: &str) -> IonResult<&str, &str> {
+    recognize(many0(ws))(i)
+}
+
+/// ws
+///     : WHITESPACE
+///     | INLINE_COMMENT
+///     | BLOCK_COMMENT
+///     ;
+fn ws(i: &str) -> IonResult<&str, &str> {
+    alt((whitespace, take_inline_comment, take_block_comment))(i)
+}
+
+///
+/// Encoding Section: Ion Punctuation
+///
+
+/// L_BRACKET : '[';
+const L_BRACKET: char = '[';
+
+/// R_BRACKET : ']';
+const R_BRACKET: char = ']';
+
+/// L_PAREN   : '(';
+const L_PAREN: char = '(';
+
+/// R_PAREN   : ')';
+const R_PAREN: char = ')';
+
+/// L_CURLY   : '{';
+const L_CURLY: char = '{';
+
+/// R_CURLY   : '}';
+const R_CURLY: char = '}';
+
+/// COMMA     : ',';
+const COMMA: char = ',';
+
+/// COLON     : ':';
+const COLON: char = ':';
+
+/// DOT       : '.';
+const DOT: char = '.';
+
+/// Note: test 'bad/sexpBackslashNL.ion' suggest that \ is not a valid operator character.
+///
+/// NON_DOT_OPERATOR
+///     : [!#%&*+\-/;<=>?@^`|~]
+///     ;
+fn is_non_dot_operator(c: char) -> bool {
+    [
+        '!', '#', '%', '&', '*', '+', '-', '/', ';', '<', '=', '>', '?', '@', '^', '`', '|', '~',
+    ]
+    .contains(&c)
+}
+
+///
+/// Encoding Section: Ion Whitespace / Comments
+///
+
+/// Consumes whitespace (not comments). Used in LOBs.
+fn eat_whitespace<Input>(i: Input) -> IonResult<Input, Input>
+where
+    Input: Clone
+        + PartialEq
+        + InputIter
+        + InputTake
+        + InputLength
+        + InputTakeAtPosition
+        + Slice<RangeTo<usize>>
+        + Offset,
+    <Input as InputTakeAtPosition>::Item: AsChar,
+{
+    recognize(many0(whitespace))(i)
+}
```

Comment:
Also here `many0` makes me think this should be named `eat_opt_whitespace`

Reply:
Done.